litellm/proxy/proxy_server.py

```diff
@@ -171,6 +171,15 @@ def __init__(
         self.param = param
         self.code = code
 
+        # rules for proxyExceptions
+        # Litellm router.py returns "No healthy deployment available" when there are no deployments available
+        # Should map to 429 errors https://github.com/BerriAI/litellm/issues/2487
+        if (
+            "No healthy deployment available" in self.message
+            or "No deployments available" in self.message
+        ):
+            self.code = 429
```

Comment:
is the rate limit exceeded issue the only reason deployments are not available/healthy?

Reply:
> is the rate limit exceeded error the only reason deployments are not available/healthy?

No, it's not. You could enter a Bad API Key for a model and then you'll start seeing `no models available` after you send the 3rd request to the model

In my opinion: 
- Server Side we can show the cause of the error 
- Client side, when no models available user should see 429

https://github.com/BerriAI/litellm/issues/2487

Reply:
cc @gagarinfan what would you like the proxy to do here ?

Reply:
merging this, @gagarinfan feel free to open a new issue regarding this if you don't agree